### PR TITLE
Test with Python 3.13 and add PyPI trove classifiers

### PR DESCRIPTION
**Pull request by edgarrmondragon**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -27,7 +33,7 @@ jobs:
         uses: conda-incubator/setup-miniconda@v3
         with:
           environment-file: ci/env.yaml
-          python-version: ${{ matrix.PY }}
+          python-version: ${{ matrix.python-version }}
```
Comment (edgarrmondragon) [inline .github/workflows/ci.yml:36]: This was previously causing all tests to run on `conda-incubator/setup-miniconda`'s default Python (i.e. 3.12)

Comment (martindurant) [reply .github/workflows/ci.yml:36]: oops

**Comment by edgarrmondragon** [comment]
FWIW the 3.13 tests are 🟢 on my fork: https://github.com/edgarrmondragon/s3fs/actions/runs/11228720331/job/31213080263

**Comment by martindurant** [comment]
py3.13 has apparently come out, but only for `defaults` so far

**Pull request merged**